### ISSUE TITLE
[Help]: how get i deformation of vertical and Horizontal  by combining DES and ASC in DInSAR

### ISSUE THREAD

**Issue by jerry7nol** [gmtsar/gmtsar#520]
i wanna get deformation of vertical and Horizontal by combining DES and ASC in DInSAR, what steps or csh should i use

**Comment by Xiaohua-Eric-Xu** [comment]
You could 1) get asc_los.grd and des_los.grd from two tracks of measurements. 
2) use grdsample to coregister them to the same grid.
3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical and horizontal(or east) = (des_los - asc_los)/2/average_look_cosine_east. This can be done with `gmt grdmath`. 

For less simple decomposition, you may need to form an inverse problem and do things pixel-wise.

**Comment by joejoe03050709** [comment]
> You could 1) get asc_los.grd and des_los.grd from two tracks of measurements. 2) use grdsample to coregister them to the same grid. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical and horizontal(or east) = (des_los - asc_los)/2/average_look_cosine_east. This can be done with .`gmt grdmath`
> 
> For less simple decomposition, you may need to form an inverse problem and do things pixel-wise.

thank you very much sir, i will learn that.

**Comment by KumarKrishan** [comment]
Dear @Xiaohua-Eric-Xu! If any tutorial or reference paper available, will be helpful.

**Comment by SteffanDavies** [comment]
I'm surprised GMTSAR doesn't have this tool already. I have a moving box filter with a custom cell size in metres which aggregates points of the same orbit onto a grid using custom CRS. Unfortunately it's incredibly slow and needs parallelizing.

**Comment by Xiaohua-Eric-Xu** [comment]
The more complex formula will take longer especially if you account for variations in look angle. The simple formula is just one line of gmt code so we never thought of writing it up. Problem with this type of decomposition is that one has to make certain assumptions (e.g. north component could be neglected, etc.), which may not always be the case.

**Comment by SteffanDavies** [comment]
> The more complex formula will take longer especially if you account for variations in look angle. The simple formula is just one line of gmt code so we never thought of writing it up. Problem with this type of decomposition is that one has to make certain assumptions (e.g. north component could be neglected, etc.), which may not always be the case.

Changing local incidence angle can be obtained from SNAP very easily by exporting this product during range-doppler geometric correction with DEM. It's aa geolocated IMG/HDR file.

**Comment by Xiaohua-Eric-Xu** [comment]
> Changing local incidence angle can be obtained from SNAP very easily by exporting this product during range-doppler geometric correction with DEM. It's a geolocated IMG/HDR file.

That's not the time consuming part I think (which is easy to do as you mentioned). The time consuming part is the inversion when you have to update Green's matrix I think.

**Comment by shivamthescienceguy** [comment]
> You could 1) get asc_los.grd and des_los.grd from two tracks of measurements. 2) use grdsample to coregister them to the same grid. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical and horizontal(or east) = (des_los - asc_los)/2/average_look_cosine_east. This can be done with `gmt grdmath`.
> 
> For less simple decomposition, you may need to form an inverse problem and do things pixel-wise.

Could be please elaborate these steps in simple manner as to how to resample with examples, cause most of us are new to GMT.

Thanks Eric for all the help.

**Comment by shivamthescienceguy** [comment]
and where do I get the value of _average_look_cosine_vertical_ in this regard?
Waiting for your kind response.

**Comment by Xiaohua-Eric-Xu** [comment]
GMT is a really really useful tool and their developers provide short course on annual basis so you are welcome to participate that. 

e.g. ```
gmt grdmath asc.grd des.grd ADD 2 DIV $cos_v DIV = vert.grd
gmt grdmath des.grd asc.grd SUB 2 DIV $cos_e DIV = hori.grd
```

> and where do I get the value of _average_look_cosine_vertical_ in this regard?
> Waiting for your kind response.

I would just pick the point in the middle of your AOI and run SAT_look.

**Comment by Xiaohua-Eric-Xu** [comment]
GMT is a really really useful tool and their developers provide short course on annual basis so you are welcome to participate that. 

e.g. ```
gmt grdmath asc.grd des.grd ADD 2 DIV $cos_v DIV = vert.grd
gmt grdmath des.grd asc.grd SUB 2 DIV $cos_e DIV = hori.grd
```

> and where do I get the value of _average_look_cosine_vertical_ in this regard?
> Waiting for your kind response.

I would just pick the point in the middle of your AOI and run SAT_look.

**Comment by shivamthescienceguy** [comment]
Thanks alot Eric, yeah I would surely like to attend that one.
Thanks again this clears up alot of doubts

**Comment by omarwater98** [comment]
.

**Comment by omarwater98** [comment]
> You could 1) get asc_los.grd and des_los.grd from two tracks of measurements. 2) use grdsample to coregister them to the same grid. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical and horizontal(or east) = (des_los - asc_los)/2/average_look_cosine_east. This can be done with `gmt grdmath`.
> 
> For less simple decomposition, you may need to form an inverse problem and do things pixel-wise.



> You could 1) get asc_los.grd and des_los.grd from two tracks of measurements. 2) use grdsample to coregister them to the same grid. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical and horizontal(or east) = (des_los - asc_los)/2/average_look_cosine_east. This can be done with `gmt grdmath`.
> 
> For less simple decomposition, you may need to form an inverse problem and do things pixel-wise.

![Flujo de trabajo de un interferograma complejo GMTSAR_page-0001](https://user-images.githubusercontent.com/113547446/215806446-0f5356bc-607e-4c5f-9d7a-4e03868760ea.jpg)

I would have to apply the alignment to my stack of ascending images and at the same time to my stack of descending images with:
   preproc_batch_tops.csh dem.grd 2 &> pbt_mode1.log &

Run the intf_tops_parallel.csh intf.in batch_tops.config 6 &> itp.log &

and then use :

gmt grdsample phase.grd -Gnew_phase.grd

??

**Comment by omarwater98** [comment]
> @Xiaohua-Eric-Xu
> 
> > Podría 1) obtener asc_los.grd y des_los.grd de dos pistas de medidas. 2) use grdsample para registrarlos en la misma cuadrícula. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical y horizontal(u east) = (des_los - asc_los)/2/average_look_cosine_east. Esto se puede hacer con `gmt grdmath`.
> > Para una descomposición menos simple, es posible que deba formar un problema inverso y hacer las cosas en píxeles.
> 
> > Podría 1) obtener asc_los.grd y des_los.grd de dos pistas de medidas. 2) use grdsample para registrarlos en la misma cuadrícula. 3) use vertical = (asc_los + des_los)/2/average_look_cosine_vertical y horizontal(u east) = (des_los - asc_los)/2/average_look_cosine_east. Esto se puede hacer con `gmt grdmath`.
> > Para una descomposición menos simple, es posible que deba formar un problema inverso y hacer las cosas en píxeles.
> 
> ![Flujo de trabajo de un interferograma complejo GMTSAR_page-0001](https://user-images.githubusercontent.com/113547446/215806446-0f5356bc-607e-4c5f-9d7a-4e03868760ea.jpg)
> 
>I would have to do the complete time series for both type of images DES and ASC?
> 
> > y luego usar:
> 
> gmt grdsample
> 
> ??

**Comment by Xiaohua-Eric-Xu** [comment]
> I would have to do the complete time series for both type of images DES and ASC?
> 
> > 
> 
> And then to use:
> 
> > 
> 
> gmt grdsample
> 
> > 
> 
> ??

I am not exactly sure what you would like to do here. Are you trying to do a decomposition? If so, yes then.

**Comment by CUMT530849827** [comment]
@Xiaohua-Eric-Xu 
Hello Professor, I obtained the NEU of the ascending orbit when doing deformation decomposition, and calculated the east-west component and vertical component with grdmath, but I found that there was an error when calculating the east-west component, I don't know if $cos_e is given in this way.

2. In addition, I would like to know if there are any videos or documents about GMTSAR parameters. I want to learn more about GMTSAR

![图片](https://github.com/gmtsar/gmtsar/assets/49933485/428232e5-67c0-4ed1-a684-5a829c797daa)
